konnect-runtime-setup.sh

```diff
@@ -192,98 +183,20 @@ http_res_body() {
     echo "$@" | sed -e 's/HTTP_STATUS_CODE\:.*//g'
 }
 
-# login to the Konnect and acquire the session
-login() {
-    log_debug "=> entering login phase"
-
-    ARGS="--cookie-jar ./$KONNECT_HTTP_SESSION_NAME -X POST -d {\"username\":\"$KONNECT_USERNAME\",\"password\":\"$KONNECT_PASSWORD\"} --url $KONNECT_API_URL/kauth/api/v1/authenticate"
-    if [[ $KONNECT_DEV -eq 1 ]]; then
-        ARGS="-u $KONNECT_DEV_USERNAME:$KONNECT_DEV_PASSWORD $ARGS"
-    fi
-
-    RES=$(http_req "$ARGS")
-    STATUS=$(http_status "$RES")
-
-    if ! [[ $STATUS -eq 200 ]]; then
-        log_debug "==> response retrieved: $RES"
-        error "login to Konnect failed... (Status code: $STATUS)"
-    fi
-    log_debug "=> login phase completed"
-}
-
-get_control_plane() {
-    log_debug "=> entering control plane metadata retrieval phase"
+verify_certificates() {
+    log_debug "=> entering certificate verification phase"
 
-    ARGS="--cookie ./$KONNECT_HTTP_SESSION_NAME -X GET --url $KONNECT_API_URL/api/runtime_groups/$KONNECT_CONTROL_PLANE"
-    if [[ $KONNECT_DEV -eq 1 ]]; then
-        ARGS="-u $KONNECT_DEV_USERNAME:$KONNECT_DEV_PASSWORD $ARGS"
-    fi
-
-    log_debug "$ARGS"
-
-    RES=$(http_req_plain "$ARGS")
-    RESPONSE_BODY=$(http_res_body "$RES")
-    STATUS=$(http_status "$RES")
-
-    log_debug "$RESPONSE_BODY"
-
-    if [[ $STATUS -eq 200 ]]; then
-        CONTROL_PLANE=$(echo "$RESPONSE_BODY" | jq .)
-        KONNECT_CP_ID=$(echo "$CONTROL_PLANE" | jq -r .id)
-        KONNECT_CP_NAME=$(echo "$CONTROL_PLANE" | jq -r .name)
-        KONNECT_CP_ENDPOINT="$(echo "$CONTROL_PLANE" | jq -r .config.cp_outlet)"
-        KONNECT_TP_ENDPOINT="$(echo "$CONTROL_PLANE" | jq -r .config.telemetry_endpoint)"
-    else 
-        log_debug "==> response retrieved: $RES"
-        error "failed to fetch control plane (Status code: $STATUS)"
-    fi
-    log_debug "=> control plane metadata retrieval phase completed"
-}
-
-generate_certificates() {
-    log_debug "=> entering certificate generation phase"
-
-    tee -a openssl.cnf << EOF 
-[ req ]
-prompt                 = no
-distinguished_name     = req_distinguished_name
-req_extensions         = v3_req
-
-
-[ req_distinguished_name ]
-countryName            = US
-commonName             = kongdp
+    printf "%b" "$KONNECT_CERTIFICATE_KEY" > cluster.key
```

Comment:
minor:

since we override and clean up the "cluster.key" and "cluster.crt", for edge case that the caller have the same file in the same file which important. let's add an extract guard logic before the `check_variables`, for checking the $PWD does not have these 2 files.

Reply:
Updated logic to append unique suffix to generated key and certificate files.